### PR TITLE
feat(crons): Add deletion apis for processing errors

### PR DESCRIPTION
**Pull request by wedamija**
This adds apis at both the monitor and project level for deleting checkin processing errors.

### CODE REVIEW

**Comment by wedamija** [comment]
If we want, we could just make this api per project as well, instead of having two separate ones

File: src/sentry/monitors/endpoints/project_processing_errors_details.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            403: RESPONSE_FORBIDDEN,
+            404: RESPONSE_NOT_FOUND,
+        },
+    )
+    def delete(self, request: Request, project: Project, uuid: str) -> Response:
+        try:
+            parsed_uuid = UUID(uuid)
+        except ValueError:
+            raise ValidationError("Invalid UUID")
+        CheckinProcessErrorsManager().delete_for_project(project, parsed_uuid)
```
Comment (evanpurkhiser) [inline src/sentry/monitors/endpoints/project_processing_errors_details.py:56]: I am starting to feel like maybe this should just be in it's own module lol nbd

**Comment by evanpurkhiser** [comment]
> If we want, we could just make this api per project as well, instead of having two separate ones

could add a query parameter to filter to monitor slug?

File: src/sentry/monitors/endpoints/project_processing_errors_details.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/monitors/endpoints/project_processing_errors_details.py:56]: Yeah... Feels weird to instantiate it. What if I just make all the methods class/static?

Comment (evanpurkhiser) [reply src/sentry/monitors/endpoints/project_processing_errors_details.py:56]: sgtm. I am good w/ a module too

**Comment by wedamija** [comment]
> > If we want, we could just make this api per project as well, instead of having two separate ones
> 
> could add a query parameter to filter to monitor slug?

I can delete it without the monitor slug anyway. We only really have to delete the actually error, we don't have to delete from the index since it'll just drop any errors that don't have values. But I think I can also just get the monitor info from the stores processing error. Maybe it's better to just do that?

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**